Firestore/Source/Public/FIRDocumentReference.h

```diff
@@ -197,6 +198,9 @@ NS_SWIFT_NAME(DocumentReference)
 - (void)getDocumentWithCompletion:(FIRDocumentSnapshotBlock)completion
     NS_SWIFT_NAME(getDocument(completion:));
 
+- (void)getDocumentWithCompletion:(FIRDocumentSnapshotBlock)completion getOptions:(FIRGetOptions *)getOptions
```

Comment:
Completion handlers need to the last argument. This makes things work nicely both in Objective-C wherein you don't end up with labels after the block you pass and in Swift where you can use trailing closure syntax.

On DocumentSnapshot we did getDataWithOptions where the swift name was getData(with:). The analogy for this API would be

```
- (void)getDocumentWithOptions:(FIRGetOptions *)options
                    completion:(FIRDocumentSnapshotBlock)completion 
    NS_SWIFT_NAME(getDocument(with:completion:)
```

clang-format will actually choke on this double-colon thing too so you'll need to hand format and disable clang-format for the section. See `[FIRCollectionReference addDocumentWithData:completion:]` for an example of what this looks like fully formed.

Reply:
Done.